mne/viz/backends/_qt.py

```diff
@@ -13,39 +14,758 @@
+
+        if scroll is not None:
+            self._scroll_widget = QWidget()
+            self._parent_layout = QHBoxLayout()
+            self._parent_layout.addWidget(
+                _ScrollArea(scroll[0], scroll[1], self._scroll_widget))
+            self._scroll_widget.setLayout(self)
+
+        self._height = height
+
+    def _add_widget(self, widget):
+        """Add a widget to an existing layout."""
+        if isinstance(widget, QLayout):
+            self.addLayout(widget)
+        else:
+            if self._height is not None:
+                widget.setMinimumHeight(self._height)
+                widget.setMaximumHeight(self._height)
+            self.addWidget(widget)
+
+    def _add_stretch(self, amount=1):
+        self.addStretch(amount)
+
+
+class _VBoxLayout(QVBoxLayout, _AbstractVBoxLayout, _Widget,
+                  metaclass=_BaseWidget):
+
+    def __init__(self, width=None, scroll=None):
+        _AbstractVBoxLayout.__init__(self, width=width, scroll=scroll)
+        _Widget.__init__(self)
+        QVBoxLayout.__init__(self)
+
+        if scroll is not None:
+            self._scroll_widget = QWidget()
+            self._parent_layout = QHBoxLayout()
+            self._parent_layout.addWidget(
+                _ScrollArea(scroll[0], scroll[1], self._scroll_widget))
+            self._scroll_widget.setLayout(self)
+
+        self._width = width
+
+    def _add_widget(self, widget):
+        """Add a widget to an existing layout."""
+        if isinstance(widget, QLayout):
+            self.addLayout(widget)
+        else:
+            if self._width is not None:
+                widget.setMinimumWidth(self._width)
+                widget.setMaximumWidth(self._width)
+            self.addWidget(widget)
+
+    def _add_stretch(self, amount=1):
+        self.addStretch(amount)
+
+
+class _GridLayout(QGridLayout, _AbstractGridLayout, _Widget,
+                  metaclass=_BaseWidget):
+
+    def __init__(self, height=None, width=None):
+        _AbstractGridLayout.__init__(self)
+        _Widget.__init__(self)
+        QGridLayout.__init__(self)
+        if height:
+            self.setMinimumHeight(height)
+            self.setMaximumHeight(height)
+        if width:
+            self.setMinimumWidth(width)
+            self.setMaximumWidth(width)
+
+    def _add_widget(self, widget, row=None, col=None):
+        """Add a widget to an existing layout."""
+        if isinstance(widget, QLayout):
+            self.addLayout(widget, row, col)
+        else:
+            self.addWidget(widget, row, col)
+
+
+class _BaseCanvas(type(FigureCanvas), type(_AbstractCanvas)):
+    pass
+
+
+class _Canvas(FigureCanvas, _AbstractCanvas, metaclass=_BaseCanvas):
+
+    def __init__(self, width, height, dpi):
+        _AbstractCanvas.__init__(
+            self, width=width, height=height, dpi=dpi)
+        self.fig = Figure(figsize=(width, height), dpi=dpi)
+        self.ax = self.fig.add_subplot(111, position=[0.15, 0.15, 0.75, 0.75])
```

Comment:
just curious where these position values come from, are they arbitrary? tailored to a specific kind of inset plot (i.e. the traces from clicked verts in a Brain GUI)?  Wondering if we'll regret hard-coding this later, if some other plot type gets inserted here (e.g., I can easily imagine a time-freq image plot making sense in a Brain GUI, but potentially any others?)